bindings/test/config.test.ts

```diff
@@ -13,8 +15,50 @@ suite('config', () => {
     expect(() => duckdb.get_config_flag(-1)).toThrowError(/^Config option not found$/);
   });
   test('create, set, and destroy', () => {
-    const config = duckdb.create_config()
+    const config = duckdb.create_config();
     expect(config).toBeTruthy();
     duckdb.set_config(config, 'custom_user_agent', 'my_user_agent');
   });
+  test('default duckdb_api without explicit config', async () => {
+    const db = await duckdb.open();
+    const connection = await duckdb.connect(db);
+    const result = await duckdb.query(connection, `select current_setting('duckdb_api') as duckdb_api`);
+      await expectResult(result, {
+        columns: [
+          { name: 'duckdb_api', logicalType: { typeId: duckdb.Type.VARCHAR } },
+        ],
+        chunks: [
+          { rowCount: 1, vectors: [data(16, [true], ['node-neo-bindings'])]},
+        ],
+      });
+  });
+  test('default duckdb_api with explicit config', async () => {
+    const config = duckdb.create_config();
+    const db = await duckdb.open(undefined, config);
+    const connection = await duckdb.connect(db);
+    const result = await duckdb.query(connection, `select current_setting('duckdb_api') as duckdb_api`);
+      await expectResult(result, {
+        columns: [
+          { name: 'duckdb_api', logicalType: { typeId: duckdb.Type.VARCHAR } },
+        ],
+        chunks: [
+          { rowCount: 1, vectors: [data(16, [true], ['node-neo-bindings'])]},
+        ],
+      });
+  });
+  test('overriding duckdb_api', async () => {
```

Comment:
overriding duckdb_api is not really expected (although possible). All layers above the closest-to-duckdb are expected to set `custom_user_agent` instead, which is additive (all layers will be preserved).

Reply:
I needed to make at least the bindings layer overridable, so the api layer can override it. But I agree it's unclear whether the ability to override this at the api level is useful, even though I support it for completeness.